supertokens_python/recipe/thirdpartyemailpassword/emaildelivery/services/backward_compatibility/__init__.py

```diff
@@ -0,0 +1,81 @@
+from supertokens_python.recipe.emailpassword.utils import \
+    InputEmailVerificationConfig as EPInputEmailVerificationConfig
+from supertokens_python.recipe.emailpassword.utils import \
+    InputResetPasswordUsingTokenFeature
+from supertokens_python.recipe.thirdpartyemailpassword.recipeimplementation.implementation import \
+    RecipeInterface
+from supertokens_python.recipe.thirdpartyemailpassword.types import \
+    TypeThirdPartyEmailPasswordEmailDeliveryInput
+from supertokens_python.supertokens import AppInfo
+
+if TYPE_CHECKING:
+    from supertokens_python.recipe.thirdpartyemailpassword.utils import \
+        InputEmailVerificationConfig
+
+
+class BackwardCompatibilityService(EmailDeliveryInterface[TypeThirdPartyEmailPasswordEmailDeliveryInput]):
+    app_info: AppInfo
+    ep_backward_compatiblity_service: EPBackwardCompatibilityService
+
+    def __init__(self,
+                 app_info: AppInfo,
+                 recipe_interface_impl: RecipeInterface,
+                 ep_recipe_interface_impl: EPRecipeInterface,
+                 reset_password_using_token_feature: Union[InputResetPasswordUsingTokenFeature, None] = None,
+                 email_verification_feature: Union[InputEmailVerificationConfig, None] = None,
+                 ) -> None:
+        self.recipe_interface_impl = recipe_interface_impl
+
+        ep_bc_email_verification_feature = None
+        if email_verification_feature is not None and email_verification_feature.create_and_send_custom_email is not None:
+            # TODO: Shouldn't it be `user: EVUser`??
+            async def create_and_send_custom_email_wrapper(user: EPUser, link: str, user_context: Dict[str, Any]):
+                user_info = await recipe_interface_impl.get_user_by_id(user.user_id, user_context)
+                if user_info is None:
+                    raise Exception("Unknown User ID provided")
+
+                assert email_verification_feature.create_and_send_custom_email
+                # TODO: WHY IS THE CODE NOT WORKING WITHOUT THIS ASSERT?? (UNEXPECTED BEHAVIOUR)
+
+                return await email_verification_feature.create_and_send_custom_email(user_info, link, user_context)
+
+            ep_bc_email_verification_feature = EPInputEmailVerificationConfig(
+                None,  # TODO: Is this okay? (Most likely no) But the node PR isn't passing get_email_verification_url
+                create_and_send_custom_email_wrapper
+            )
+
+        self.ep_backward_compatiblity_service = EPBackwardCompatibilityService(
+            app_info,
+            ep_recipe_interface_impl,
+            reset_password_using_token_feature,
+            ep_bc_email_verification_feature,
+        )
+
+    async def send_email(self, input_: TypeEmailPasswordEmailDeliveryInput, user_context: Dict[str, Any]) -> None:
+        return await self.ep_backward_compatiblity_service.send_email(input_, user_context)
```

Comment:
This has changed in the node PR. We do require email verification and email password service. Sorry for the confusion.

Reply:
Okay. Restored.